src/test/pages/EventDetails.test.tsx

```diff
@@ -306,6 +316,50 @@ describe("EventDetails", () => {
     expect(joinEventButton()).toBeInTheDocument();
     expect(joinEventButton()).toHaveAttribute("href", event.meetingUrl);
   });
+
+  it("if details are missing, 'book now' button should be invalid", async () => {
+    const event = { ...mockEvent, placesAvailable: 10 };
+    await setupTest({
+      role: "STUDENT",
+      event,
+      modifyUser: (user) =>
+        produce(user, (u) => {
+          u.givenName = undefined;
+          u.familyName = undefined;
+          u.registeredContexts = [];
+        }),
+    });
+    const bookButton = screen.getByRole("button", { name: "Book a place" });
+    await userEvent.click(bookButton);
+    const firstNameInput = screen.getByRole("textbox", { name: "First name" });
+    expect(firstNameInput).toBeInvalid();
+    const familyNameInput = screen.getByRole("textbox", { name: "Last name" });
+    expect(familyNameInput).toBeInvalid();
+    const stageInput = screen.getByRole("textbox", { name: "Stage" });
+    expect(stageInput).toBeInvalid();
+    const examBoardInput = screen.getByRole("textbox", { name: "Exam board" });
+    expect(examBoardInput).toBeInvalid();
```

Comment:
These could be done like this to be a little more concise:

`[firstNameInput, familyNameInput, stageInput, examBoardInput].forEach((each)=>expect(each).toBeInvalid())`

Reply:
Sorted